examples/platforms/posix/radio.c

```diff
@@ -576,11 +619,21 @@ void posixRadioProcess(otInstance *aInstance)
     }
 }
 
-void radioTransmit(const struct RadioMessage *msg, const struct RadioPacket *pkt)
+void radioTransmit(struct RadioMessage *msg, const struct RadioPacket *pkt)
 {
     uint32_t i;
     struct sockaddr_in sockaddr;
 
+    uint16_t crc = 0;
+    uint16_t crc_offset = pkt->mLength - sizeof(uint16_t);
+
+    for (i = 0; i < crc_offset; i++)
+    {
+        crc = crc16_citt(crc, msg->mPsdu[i]);
+    }
+
+    memcpy(&msg->mPsdu[crc_offset], &crc, sizeof(crc));
```

Comment:
Overall looks good. Thanks for addressing this.
One suggestion here: Maybe we can append CRC bytes one by one instead of `memcpy` to be safe against any endian-ness (byte order) issues.